packages/valory/skills/decision_maker_abci/behaviours/sampling.py

```diff
@@ -114,16 +155,27 @@ def _sample(self) -> Optional[int]:
             self.context.logger.info(f"Simulating date: {datetime.fromtimestamp(now)}")
         else:
             now = self.synced_timestamp
+
+        # filter out only the bets that are processable and have a queue_status that allows them to be sampled
         available_bets = list(
-            filter(lambda bet: self.processable_bet(bet, now=now), self.bets)
+            filter(
+                lambda bet: self.processable_bet(bet, now=now),
+                self.bets,
+            )
         )
         if len(available_bets) == 0:
             msg = "There were no unprocessed bets available to sample from!"
             self.context.logger.warning(msg)
             return None
 
+        # sample a bet using the priority logic
         idx = self._sampled_bet_idx(available_bets)
-        sampled_bet = self.bets[idx]
+        if idx:
+            sampled_bet = self.bets[idx]
+        else:
+            return None
```

Comment:
I wonder why `_sampled_bet_idx`'s return value is optional. Can it return `None`?

Reply:
it can never return None, as `_sample` function only sets `available_bets` list when it's not empty, logically we can remove it